### ISSUE TITLE
lint for unnecessary closures

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#3238]
**Describe the rule you'd like to see implemented**
Lint if somebody instantiates a closure that just contains one method call and forwards all its parameters to that method call.

**Examples**
Bad:
```dart
OverlayEntry(builder: (BuildContext context) => _buildStartHandle(context));
```

Good:
```dart
OverlayEntry(builder: _buildStartHandle);
```

**Additional context**
See example in https://github.com/flutter/flutter/pull/98409/files.

**Comment by bwilkerson** [comment]
Does [`unnecessary_lambdas`](https://dart-lang.github.io/linter/lints/unnecessary_lambdas.html) do what you want?

**Comment by tgucio** [comment]
Indeed, unnecessary_lambdas is what caught the lambdas from #98409.

**Comment by goderbauer** [comment]
Ah, I missed that one. And now I also remember why we don't have that one enabled: There are cases where we explicitly do want the unnecessary lambda (see https://github.com/dart-lang/linter/issues/498), but the lint doesn't have a way for a method-side opt-out (e.g. to say that the lint should never apply to anything passed to `setState`...

**Comment by pq** [comment]
Closing this in favor of picking up the conversation around exceptions to `unnecessary_lambdas` in #498.
